plugins/inputs/whois/whois_test.go

```diff
@@ -0,0 +1,198 @@
+				Status:               []string{"clientTransferProhibited"},
+				NameServers:          []string{"ns1.example.com", "ns2.example.com"},
+			},
+			Registrar: &whoisparser.Contact{
+				Name: "RESERVED-Internet Assigned Numbers Authority",
+			},
+		},
+	}
+
+	plugin.whoisLookup = func(_ *whois.Client, domain string, _ string) (string, error) {
+		return "WHOIS mock response for " + domain, nil
+	}
+
+	plugin.parseWhoisData = func(raw string) (whoisparser.WhoisInfo, error) {
+		for domain, info := range mockResponses {
+			if strings.Contains(raw, domain) { // Match requested domain
+				return info, nil
+			}
+		}
+
+		return whoisparser.WhoisInfo{}, errors.New("mock WHOIS data not found")
+	}
+
+	require.NoError(t, plugin.Gather(acc))
+
+	require.Equal(t, "example.com", acc.TagValue("whois", "domain"))
+	domainStatus, found := acc.IntField("whois", "status_code")
+	require.True(t, found, "Expected field status_code not found")
+	require.Equal(t, int(3), domainStatus, "Expected status_code field mismatch")
+
+	// Validate `expiration_timestamp` field (2025-08-13T04:00:00Z → Unix)
+	expectedExpiration := int64(1755057600)
+	expirationValue, found := acc.Int64Field("whois", "expiration_timestamp")
+	require.True(t, found, "expiration_timestamp field missing")
+	require.InDelta(t, expectedExpiration, expirationValue, 10)
+
+	// Validate `expiry` field
+	now := time.Now()
+	expectedExpiry := int(expectedExpiration - now.Unix())
+	expiryValue, found := acc.IntField("whois", "expiry")
+	require.True(t, found, "expiry field missing")
+	require.InDelta(t, expectedExpiry, expiryValue, 10) // Allow small delta due to execution time
+}
+
+// Test WHOIS Handling for an Invalid Domain
+func TestWhoisGatherInvalidDomain(t *testing.T) {
+	plugin := &Whois{
+		Domains: []string{"invalid-domain.xyz"},
+		Log:     testutil.Logger{},
+	}
+
+	require.NoError(t, plugin.Init(), "Unexpected error during Init()")
+	acc := &testutil.Accumulator{}
+
+	// Mock `whoisLookup` to return a valid WHOIS response (not an error)
+	plugin.whoisLookup = func(_ *whois.Client, _ string, _ string) (string, error) {
+		return "WHOIS mock response for invalid-domain.xyz", nil
+	}
+
+	// Mock `parseWhoisData` to return ErrNotFoundDomain (simulating WHOIS parser failure)
+	plugin.parseWhoisData = func(_ string) (whoisparser.WhoisInfo, error) {
+		return whoisparser.WhoisInfo{}, whoisparser.ErrNotFoundDomain
+	}
+
+	err := plugin.Gather(acc)
+	require.NoError(t, err)
+
+	// Ensure the metric is recorded
+	require.True(t, acc.HasMeasurement("whois"))
+
+	// Validate domain tag
+	domainTag := acc.TagValue("whois", "domain")
+	require.Equal(t, "invalid-domain.xyz", domainTag, "Expected domain tag mismatch")
+
+	// Validate status_code is set to 6 (ErrNotFoundDomain)
+	statusCode, found := acc.IntField("whois", "status_code")
+	require.True(t, found, "Expected status_code field missing")
+	require.Equal(t, 6, statusCode, "Expected status_code to be 6 for NotFoundDomain error")
```

Comment:
This can be replaced with internal tooling:

```suggestion
	testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics(), testutil.IgnoreTime())
```

Reply:
Fixed.